qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,591 @@
+    _type = _interaction_type(n_number, n_raise, n_lower)
+    # For EXCITATION and NUMBER_EXCITATION, create an edge between the `+` and `-`.
+    if _type in (TermType.EXCITATION, TermType.NUMBER_EXCITATION):
+        inds = [i for (i, c) in facs if c in "+-"]
+        if len(inds) != 2:
+            raise ValueError("wrong number or raising and lowering: ", len(inds))
+        _add_one_edge(edge_matrix, *inds)
+    # For `double_excitation` create an edge between the two `+`s and an edge between the two `-`s.
+    elif _type == TermType.DOUBLE_EXCITATION:
+        raise_inds = [i for (i, c) in facs if c == "+"]
+        lower_inds = [i for (i, c) in facs if c == "-"]
+        _add_one_edge(edge_matrix, *raise_inds)
+        _add_one_edge(edge_matrix, *lower_inds)
+
+
+def _bksf_edge_list_fermionic_op(ferm_op: FermionicOp) -> np.ndarray:
+    """Construct edge list required for the BKSF algorithm.
+
+    Args:
+        ferm_op: the fermionic operator in the second quantized form
+
+    Returns:
+        numpy.ndarray: edge_list, a 2xE matrix, where E is total number of edges
+                        and each pair denotes (from, to)
+    """
+    edge_matrix = _get_adjacency_matrix(ferm_op)
+    edge_list_as_2d_array = np.asarray(np.nonzero(edge_matrix))
+    return edge_list_as_2d_array
+
+
+def _edge_operator_aij(edge_list: np.ndarray, i: int, j: int) -> SparsePauliOp:
+    """Calculate the edge operator A_ij.
+
+    The definitions used here are consistent with arXiv:quant-ph/0003137
+
+    Args:
+        edge_list: a 2xE matrix, where E is total number of edges
+                                    and each pair denotes (from, to)
+        i: specifying the edge operator A
+        j: specifying the edge operator A
+
+    Returns:
+        Pauli: qubit operator
+    """
+    v = np.zeros(edge_list.shape[1])
+    w = np.zeros(edge_list.shape[1])
+
+    position_ij = -1
+
+    for edge_index in range(edge_list.shape[1]):
+        if set((i, j)) == set(edge_list[:, edge_index]):
+            position_ij = edge_index
+            break
+
+    w[position_ij] = 1
+
+    def _set_edges(index1, index2):
+        qubit_position = np.asarray(np.where(edge_list == index1))
+        for edge_index in range(qubit_position.shape[1]):
+            i_i, j_j = qubit_position[:, edge_index]
+            i_i = int(not i_i)
+            if edge_list[i_i][j_j] < index2:
+                v[j_j] = 1
+
+    _set_edges(i, j)
+    _set_edges(j, i)
+
+    qubit_op = Pauli((v, w))
+    return SparsePauliOp(qubit_op)
+
+
+def _edge_operator_bi(edge_list: np.ndarray, i: int) -> SparsePauliOp:
+    """Calculate the edge operator B_i.
+
+    The definitions used here are consistent with arXiv:quant-ph/0003137
+
+    Args:
+        edge_list: a 2xE matrix, where E is total number of edges
+                                    and each pair denotes (from, to)
+        i: index for specifying the edge operator B.
+
+    Returns:
+        Pauli: qubit operator
```

Comment:
Similar comment to above

Reply:
See my comments above. I did the same revision here.